src/components/TargetForm/TargetFormContent.js

```diff
@@ -69,65 +71,66 @@ const TargetFormContent = ({
     topicsError,
     loadingTopics,
     topicPlaceholder,
+    visible,
+    initialValues,
   });
 
+  const onEndEditingArea = () => {
+    setValues({
+      ...values,
+      areaLength: values.areaLength ? `${values.areaLength} m` : initialValues.areaLength,
```

Comment:
shouldn't the initialValue be the same as any value, without the 'm' character and add it here too?